crates/cli/src/lib.rs

```diff
@@ -139,35 +128,23 @@ async fn update(context: Context<impl Environment>) -> anyhow::Result<()> {
     // We want to detect this scenario and retry, or fail if we are unable to.
     // We do that with a few attempts.
     for _attempt in 1..=UPDATE_ATTEMPTS {
-        let configuration_file_path = context
-            .context_path
-            .join(configuration::CONFIGURATION_FILENAME);
-        let input: configuration::RawConfiguration = {
-            let configuration_file_contents =
-                read_config_file_contents(&configuration_file_path).await?;
-            serde_json::from_str(&configuration_file_contents)?
-        };
-        let output = configuration::introspect(input.clone(), &context.environment).await?;
+        let existing_configuration =
+            configuration::parse_configuration(&context.context_path).await?;
+        let output =
+            configuration::introspect(existing_configuration.clone(), &context.environment).await?;
 
         // Check that the input file did not change since we started introspecting,
-        let input_again_before_write: configuration::RawConfiguration = {
-            let configuration_file_contents =
-                read_config_file_contents(&configuration_file_path).await?;
-            serde_json::from_str(&configuration_file_contents)?
-        };
+        let input_again_before_write =
```

Comment:
Previously, this function only considered changes to specifically `configuration.json`, meaning changes to e.g. native queries in files wouldn't get caught.